androguard/core/apk/__init__.py

```diff
@@ -1501,35 +1504,256 @@ def is_androidtv(self) -> bool:
+            return None
+
+        # Prior to Android N, Android attempts to verify only the first SignerInfo. From N onwards, Android attempts
+        # to verify all SignerInfos and then picks the first verified SignerInfo.
+        min_sdk_version = self.get_min_sdk_version()
+        if min_sdk_version is None or int(min_sdk_version) < 24:  # AndroidSdkVersion.N
+            logger.info(f"minSdkVersion: {min_sdk_version} is less than 24. Getting the first signerInfo only!")
+            unverified_signer_infos_to_try = [signer_infos[0]]
+        else:
+            unverified_signer_infos_to_try = signer_infos
+
+        # Extract certificates from the PKCS7 object
+        certificates = signed_data['content']['certificates']
+        return_certificate = None
+        list_certificates_verified = []
+        for signer_info in unverified_signer_infos_to_try:
+            try:
+                matching_certificate_verified = self.verify_signer_info_against_sig_file(
+                    signed_data,
+                    certificates,
+                    signer_info,
+                    sf_object)
+            except (ValueError, TypeError, OSError, errors.SignatureError) as e:
+                logger.error(f"The following exception was raised while verifying the certificate: {e}")
+                return None  # the validation stops due to the exception raised!
+            if matching_certificate_verified is not None:
+                list_certificates_verified.append(matching_certificate_verified)
+        if not list_certificates_verified:
+            logger.error(f"minSdkVersion: {min_sdk_version}, # of SignerInfos: {len(unverified_signer_infos_to_try)}. None Verified!")
+        else:
+            return_certificate = list_certificates_verified[0]
+        return return_certificate
+
+    def verify_signer_info_against_sig_file(self, signed_data, certificates, signer_info, sf_object):
+        matching_certificate = self.find_certificate(certificates, signer_info)
+        matching_certificate_verified = None
+        digest_algorithm = self.get_hash_algorithm(signer_info)
+        if matching_certificate is None:
+            raise ValueError("Signing certificate referenced in SignerInfo not found in SignedData")
+        else:
+            if signer_info['signed_attrs'].native:
+                logger.info("Signed Attributes detected!")
+                signed_attrs = signer_info['signed_attrs']
+                signed_attrs_dict = OrderedDict()
+                for attr in signed_attrs:
+                    if attr['type'].dotted in signed_attrs_dict:
+                        raise ValueError(f"Duplicate signed attribute: {attr['type'].dotted}")
+                    signed_attrs_dict[attr['type'].dotted] = attr['values']
+
+                # Check content type attribute (for Android N and newer)
+                max_sdk_version = self.get_max_sdk_version()
```

Comment:
I still think using `get_max_sdk_version()` is wrong here as `apksigner` never gets this from the APK.
Having `max_sdk_version` as an optional parameter to `get_certificate_der()` would make sense though.
Perhaps `min_sdk_version` as well, defaulting to getting it from the APK, same as `apksigner`, in that case.